src/README.md

```diff
@@ -4,76 +4,103 @@ The purpose of this document is to describe the internal code structure and majo
 
-<a id="4"></a>
-### Agent
-- DAOS Agent: daos_agent is a daemon process residing on the client node. Trusted intermediary in authentication of DAOS Client application. Signs DAOS Client credentials using local certificates
+The data plane is a multi-threaded process written in C that runs the DAOS storage engine. It processes incoming metadata and I/O requests though the CART communication middleware and accesses local NVM storage via the PMDK (for storage-class memory, aka SCM) and SPDK (for NVMe SSDs) libraries. The data plane relies on Argobots for event-based parallel processing and exports multiple targets that can be independently addressed via the fabric. Each data plane instance is assigned a unique rank inside a DAOS system.
+
+The control plane and data plane processes communicate locally through Unix Domain Socket and a custom lightweight protocol called dRPC.
+
+For further reading:
+- <a href="control/README.md">DAOS control plane (daos_server)</a>
+- <a href="iosrv/README.md">DAOS data plane (daos_io_server)</a>
 
-<a href="control/agent/README.md">DAOS Agent</a>
+<a id="12"></a>
+### Client APIs, Tools and I/O Middleware
 
-<a id="5"></a>
-### Management Tool
-- DAOS Management Tool: Uses control plane client API to administrate the daos_server instances. e.g. daos_shell executable
+Applications, users and administrators can interact with a DAOS system through two different client APIs.
+
+The DAOS management Go package allows to administrate a DAOS system from any nodes that can communicate with the DAOS servers through the out-of-band management channel. This API is reserved for the DAOS system administrators who are authenticated through a specific certificate. The DAOS management API is intended to be integrated with different vendor-specific storage management or open-source orchestration frameworks. A CLI tool called dmg is built over the DAOS management API. For further reading on the management API and the dmg tool:
+- <a href="https://godoc.org/github.com/daos-stack/daos/src/control/client">DAOS management Go package</a>
+- <a href="control/dmg/README.md">DAOS Management tool (aka dmg)</a>
+
+The DAOS library (i.e. libdaos) implements the DAOS storage model and is primarily targeted at application and I/O middleware developers who want to store datasets into DAOS containers. It can be used from any nodes connected to the fabric used by the targeted DAOS system. The application process is authenticated via the DAOS agent (see next section). The API exported by libdaos is commonly called the DAOS API (in opposition to the DAOS management API) and allows to manage containers and access DAOS objects through different interfaces (e.g. key-value store or array API). The libdfs library emulates POSIX file and directory abstractions over libdaos and provides a smooth migration path for applications that requires a POSIX namespace. For further reading on libdaos, bindings for different programming languages and libdfs:
```

Comment:
sp - "requires" -> "require" - i.e. "for applications that require a POSIX namespace"